### ISSUE TITLE
Core dump in cppCodeGen tests on Ubuntu 18.04

### ISSUE THREAD

**Issue by chrisvittal** [hail-is/hail#4718]
### Hail version:
fd300f29c00349d2a9d26835e35be2b142a3505f
### What you did:
`make -C src/main/c prebuilt && ./gradlew testCppCodegen`
### What went wrong (all error messages here, including the full java stack trace):
```
testCppCodegen
:compileJava UP-TO-DATE
:generateBuildInfo
:nativeLib
tar -xzf libsimdpp-2.1.tar.gz
:compileScala UP-TO-DATE
:processResources UP-TO-DATE
:classes UP-TO-DATE
:compileTestJava UP-TO-DATE
:compileTestScala UP-TO-DATE
:processTestResources UP-TO-DATE
:testClasses UP-TO-DATE
:testCppCodegen
Running test: Test method testReadWrite(is.hail.annotations.AnnotationsSuite)

Gradle suite > Gradle test > is.hail.annotations.AnnotationsSuite.testReadWrite FAILED
    org.apache.spark.SparkException at AnnotationsSuite.scala:76
        Caused by: java.lang.AssertionError
Running test: Test method testEmptyKeys(is.hail.io.IndexSuite)

Gradle suite > Gradle test > is.hail.io.IndexSuite.testEmptyKeys FAILED
    java.lang.AssertionError at IndexSuite.scala:42
Running test: Test method testIntervalIterator(is.hail.io.IndexSuite)

Gradle suite > Gradle test > is.hail.io.IndexSuite.testIntervalIterator FAILED
    java.lang.AssertionError at IndexSuite.scala:42
Running test: Test method testIntervalIteratorWorksWithGeneralEndpoints(is.hail.io.IndexSuite)

Gradle suite > Gradle test > is.hail.io.IndexSuite.testIntervalIteratorWorksWithGeneralEndpoints FAILED
    java.lang.AssertionError at IndexSuite.scala:42
Running test: Test method testIterateFromUntil(is.hail.io.IndexSuite)

Gradle suite > Gradle test > is.hail.io.IndexSuite.testIterateFromUntil FAILED
    java.lang.AssertionError at IndexSuite.scala:42
Running test: Test method testLowerBound(is.hail.io.IndexSuite)

Gradle suite > Gradle test > is.hail.io.IndexSuite.testLowerBound FAILED
    java.lang.AssertionError at IndexSuite.scala:42
Running test: Test method testQueryByKey(is.hail.io.IndexSuite)

Gradle suite > Gradle test > is.hail.io.IndexSuite.testQueryByKey FAILED
    java.lang.AssertionError at IndexSuite.scala:42
Running test: Test method testRangeIterator(is.hail.io.IndexSuite)

Gradle suite > Gradle test > is.hail.io.IndexSuite.testRangeIterator FAILED
    java.lang.AssertionError at IndexSuite.scala:42
Running test: Test method testUpperBound(is.hail.io.IndexSuite)

Gradle suite > Gradle test > is.hail.io.IndexSuite.testUpperBound FAILED
    java.lang.AssertionError at IndexSuite.scala:42
Running test: Test method writeReadGivesSameAsInput[0]([Ljava.lang.String;@49613eb0)(is.hail.io.IndexSuite)

Gradle suite > Gradle test > is.hail.io.IndexSuite.writeReadGivesSameAsInput[0]([Ljava.lang.String;@49613eb0) FAILED
    java.lang.AssertionError at IndexSuite.scala:42
Running test: Test method writeReadGivesSameAsInput[1]([Ljava.lang.String;@326709be)(is.hail.io.IndexSuite)

Gradle suite > Gradle test > is.hail.io.IndexSuite.writeReadGivesSameAsInput[1]([Ljava.lang.String;@326709be) FAILED
    java.lang.AssertionError at IndexSuite.scala:42
Running test: Test method writeReadGivesSameAsInput[2]([Ljava.lang.String;@7c5e570e)(is.hail.io.IndexSuite)

Gradle suite > Gradle test > is.hail.io.IndexSuite.writeReadGivesSameAsInput[2]([Ljava.lang.String;@7c5e570e) FAILED
    java.lang.AssertionError at IndexSuite.scala:42
Running test: Test method writeReadGivesSameAsInput[3]([Ljava.lang.String;@3c90c5dc)(is.hail.io.IndexSuite)

Gradle suite > Gradle test > is.hail.io.IndexSuite.writeReadGivesSameAsInput[3]([Ljava.lang.String;@3c90c5dc) FAILED
    java.lang.AssertionError at IndexSuite.scala:42
Running test: Test method writeReadGivesSameAsInput[4]([Ljava.lang.String;@165d0f2d)(is.hail.io.IndexSuite)

Gradle suite > Gradle test > is.hail.io.IndexSuite.writeReadGivesSameAsInput[4]([Ljava.lang.String;@165d0f2d) FAILED
    java.lang.AssertionError at IndexSuite.scala:42
Running test: Test method writeReadGivesSameAsInput[5]([Ljava.lang.String;@2cb6cb35)(is.hail.io.IndexSuite)

Gradle suite > Gradle test > is.hail.io.IndexSuite.writeReadGivesSameAsInput[5]([Ljava.lang.String;@2cb6cb35) FAILED
    java.lang.AssertionError at IndexSuite.scala:42
Running test: Test method writeReadGivesSameAsInput[6]([Ljava.lang.String;@4c09f4c5)(is.hail.io.IndexSuite)

Gradle suite > Gradle test > is.hail.io.IndexSuite.writeReadGivesSameAsInput[6]([Ljava.lang.String;@4c09f4c5) FAILED
    java.lang.AssertionError at IndexSuite.scala:42
Running test: Test method writeReadGivesSameAsInput[7]([Ljava.lang.String;@67a910a4)(is.hail.io.IndexSuite)

Gradle suite > Gradle test > is.hail.io.IndexSuite.writeReadGivesSameAsInput[7]([Ljava.lang.String;@67a910a4) FAILED
    java.lang.AssertionError at IndexSuite.scala:42
Running test: Test method writeReadGivesSameAsInput[8]([Ljava.lang.String;@3c215dd4)(is.hail.io.IndexSuite)

Gradle suite > Gradle test > is.hail.io.IndexSuite.writeReadGivesSameAsInput[8]([Ljava.lang.String;@3c215dd4) FAILED
    java.lang.AssertionError at IndexSuite.scala:42
Running test: Test method writeReadGivesSameAsInput[9]([Ljava.lang.String;@5a40cc5)(is.hail.io.IndexSuite)

Gradle suite > Gradle test > is.hail.io.IndexSuite.writeReadGivesSameAsInput[9]([Ljava.lang.String;@5a40cc5) FAILED
    java.lang.AssertionError at IndexSuite.scala:42
Running test: Test method writeReadGivesSameAsInput[10]([Ljava.lang.String;@69594c0e)(is.hail.io.IndexSuite)

Gradle suite > Gradle test > is.hail.io.IndexSuite.writeReadGivesSameAsInput[10]([Ljava.lang.String;@69594c0e) FAILED
    java.lang.AssertionError at IndexSuite.scala:42
Running test: Test method writeReadGivesSameAsInput[11]([Ljava.lang.String;@5ea39a14)(is.hail.io.IndexSuite)

Gradle suite > Gradle test > is.hail.io.IndexSuite.writeReadGivesSameAsInput[11]([Ljava.lang.String;@5ea39a14) FAILED
    java.lang.AssertionError at IndexSuite.scala:42
Running test: Test method writeReadGivesSameAsInput[12]([Ljava.lang.String;@21ebf782)(is.hail.io.IndexSuite)

Gradle suite > Gradle test > is.hail.io.IndexSuite.writeReadGivesSameAsInput[12]([Ljava.lang.String;@21ebf782) FAILED
    java.lang.AssertionError at IndexSuite.scala:42
Running test: Test method writeReadGivesSameAsInput[13]([Ljava.lang.String;@3bda762a)(is.hail.io.IndexSuite)

Gradle suite > Gradle test > is.hail.io.IndexSuite.writeReadGivesSameAsInput[13]([Ljava.lang.String;@3bda762a) FAILED
    java.lang.AssertionError at IndexSuite.scala:42
Running test: Test method writeReadGivesSameAsInput[14]([Ljava.lang.String;@4898b38e)(is.hail.io.IndexSuite)

Gradle suite > Gradle test > is.hail.io.IndexSuite.writeReadGivesSameAsInput[14]([Ljava.lang.String;@4898b38e) FAILED
    java.lang.AssertionError at IndexSuite.scala:42
Running test: Test method writeReadGivesSameAsInput[15]([Ljava.lang.String;@51684a)(is.hail.io.IndexSuite)

Gradle suite > Gradle test > is.hail.io.IndexSuite.writeReadGivesSameAsInput[15]([Ljava.lang.String;@51684a) FAILED
    java.lang.AssertionError at IndexSuite.scala:42
Running test: Test method writeReadGivesSameAsInput[16]([Ljava.lang.String;@4a971368)(is.hail.io.IndexSuite)

Gradle suite > Gradle test > is.hail.io.IndexSuite.writeReadGivesSameAsInput[16]([Ljava.lang.String;@4a971368) FAILED
    java.lang.AssertionError at IndexSuite.scala:42
Running test: Test method writeReadGivesSameAsInput[17]([Ljava.lang.String;@47162b74)(is.hail.io.IndexSuite)

Gradle suite > Gradle test > is.hail.io.IndexSuite.writeReadGivesSameAsInput[17]([Ljava.lang.String;@47162b74) FAILED
    java.lang.AssertionError at IndexSuite.scala:42
Running test: Test method writeReadGivesSameAsInput[18]([Ljava.lang.String;@70bfbb33)(is.hail.io.IndexSuite)

Gradle suite > Gradle test > is.hail.io.IndexSuite.writeReadGivesSameAsInput[18]([Ljava.lang.String;@70bfbb33) FAILED
    java.lang.AssertionError at IndexSuite.scala:42
Running test: Test method writeReadGivesSameAsInput[19]([Ljava.lang.String;@3e391ee2)(is.hail.io.IndexSuite)

Gradle suite > Gradle test > is.hail.io.IndexSuite.writeReadGivesSameAsInput[19]([Ljava.lang.String;@3e391ee2) FAILED
    java.lang.AssertionError at IndexSuite.scala:42
Running test: Test method testCXXCodeFunctions(is.hail.nativecode.NativeCodeSuite)
DEBUG: Logging set_test_msg ...

Gradle suite > Gradle test > is.hail.nativecode.NativeCodeSuite.testCXXCodeFunctions PASSED
Running test: Test method testNativeBuild(is.hail.nativecode.NativeCodeSuite)

Gradle suite > Gradle test > is.hail.nativecode.NativeCodeSuite.testNativeBuild PASSED
Running test: Test method testNativeCallSpeed(is.hail.nativecode.NativeCodeSuite)

Gradle suite > Gradle test > is.hail.nativecode.NativeCodeSuite.testNativeCallSpeed PASSED
Running test: Test method testNativeGlobal(is.hail.nativecode.NativeCodeSuite)

Gradle suite > Gradle test > is.hail.nativecode.NativeCodeSuite.testNativeGlobal PASSED
Running test: Test method testNativePtr(is.hail.nativecode.NativeCodeSuite)

Gradle suite > Gradle test > is.hail.nativecode.NativeCodeSuite.testNativePtr PASSED
Running test: Test method testNativeUpcall(is.hail.nativecode.NativeCodeSuite)
DEBUG: Logging set_test_msg ...

Gradle suite > Gradle test > is.hail.nativecode.NativeCodeSuite.testNativeUpcall PASSED
Running test: Test method testObjectArray(is.hail.nativecode.NativeCodeSuite)

Gradle suite > Gradle test > is.hail.nativecode.NativeCodeSuite.testObjectArray PASSED
Running test: Test method testShuffleAndJoinDoesntMemoryLeak(is.hail.expr.ir.TableIRSuite)

Gradle suite > Gradle test > is.hail.expr.ir.TableIRSuite.testShuffleAndJoinDoesntMemoryLeak PASSED
Running test: Test method testBufferWriteReadDoubles(is.hail.annotations.UnsafeSuite)

Gradle suite > Gradle test > is.hail.annotations.UnsafeSuite.testBufferWriteReadDoubles PASSED
Running test: Test method testCodec(is.hail.annotations.UnsafeSuite)
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007fe4a85738ec, pid=23790, tid=0x00007fe48cdfa700
#
# JRE version: OpenJDK Runtime Environment (8.0_181-b13) (build 1.8.0_181-8u181-b13-0ubuntu0.18.04.1-b13)
# Java VM: OpenJDK 64-Bit Server VM (25.181-b13 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# J 9008 C1 is.hail.annotations.UnsafeRow$.readBinary(Lis/hail/annotations/Region;J)[B (39 bytes) @ 0x00007fe4a85738ec [0x00007fe4a8573600+0x2ec]
#
# Core dump written. Default location: /home/BROAD.MIT.EDU/cvittal/src/hail/hail/core or core.23790 (max size 9223372036854775 kB). To ensure a full core dump, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /home/BROAD.MIT.EDU/cvittal/src/hail/hail/hs_err_pid23790.log
Compiled method (c1)   33969 8500       2       is.hail.annotations.UnsafeRow$::readLocus (78 bytes)
 total in heap  [0x00007fe4a8b81810,0x00007fe4a8b83430] = 7200
 relocation     [0x00007fe4a8b81938,0x00007fe4a8b81a98] = 352
 main code      [0x00007fe4a8b81aa0,0x00007fe4a8b82100] = 1632
 stub code      [0x00007fe4a8b82100,0x00007fe4a8b822b8] = 440
 oops           [0x00007fe4a8b822b8,0x00007fe4a8b822c0] = 8
 metadata       [0x00007fe4a8b822c0,0x00007fe4a8b82338] = 120
 scopes data    [0x00007fe4a8b82338,0x00007fe4a8b82f30] = 3064
 scopes pcs     [0x00007fe4a8b82f30,0x00007fe4a8b83340] = 1040
 dependencies   [0x00007fe4a8b83340,0x00007fe4a8b83348] = 8
 nul chk table  [0x00007fe4a8b83348,0x00007fe4a8b83430] = 232
#
FATAL: caught signal 6 SIGABRT
# If you would like to submit a bug report, please visit:
#   http://bugreport.sun.com/bugreport/
#
/tmp/libhail8122447512081932366.so(+0x18f5f)[0x7fe3a7bf0f5f]
/lib/x86_64-linux-gnu/libc.so.6(+0x3ef20)[0x7fe4be507f20]
/lib/x86_64-linux-gnu/libc.so.6(gsignal+0xc7)[0x7fe4be507e97]
/lib/x86_64-linux-gnu/libc.so.6(abort+0x141)[0x7fe4be509801]
/usr/lib/jvm/java-8-openjdk-amd64/jre/lib/amd64/server/libjvm.so(+0x8e80b9)[0x7fe4bd7f00b9]
/usr/lib/jvm/java-8-openjdk-amd64/jre/lib/amd64/server/libjvm.so(+0xaaed23)[0x7fe4bd9b6d23]
/usr/lib/jvm/java-8-openjdk-amd64/jre/lib/amd64/server/libjvm.so(JVM_handle_linux_signal+0x1b4)[0x7fe4bd7fa694]
/usr/lib/jvm/java-8-openjdk-amd64/jre/lib/amd64/server/libjvm.so(+0x8e5318)[0x7fe4bd7ed318]
/lib/x86_64-linux-gnu/libc.so.6(+0x3ef20)[0x7fe4be507f20]
[0x7fe4a85738ec]

38 tests completed, 29 failed
:testCppCodegen FAILED
```
[hs_err_pid23790.log](https://github.com/hail-is/hail/files/2540933/hs_err_pid23790.log)

**Comment by chrisvittal** [comment]
The core itself is on my computer.

**Comment by catoverdrive** [comment]
@chrisvittal can you test again on the changes in #4809? (either on that branch or on master once it goes in?)

**Comment by catoverdrive** [comment]
@chrisvittal the fix for this should be in master as of #4847. When you get a chance, can you verify + close this issue (or let me know if it doesn't fix it so I can dig further)?

**Comment by chrisvittal** [comment]
I haven't run into issues since updating a few days ago. Closing.
